katcp/test/test_resource_client.py

```diff
@@ -310,12 +297,19 @@ def make_test_sensors(sensors_info):
 
         # Now get all the sensors
         result = yield DUT.list_sensors('')
-        expected_result = sorted(resource.SensorResultTuple(
-            test_sensors[s_id], test_sensors_info[s_id].name,
-            s_id, test_sensors_info[s_id].description, 'integer', '',
-            test_sensors[s_id].reading)
-                                 for s_id in test_sensors_info)
-        self.assertEqual(sorted(result), expected_result)
+        # built-in `sorted()` uses __cmp__ to sort out the order in Python2.
+        # However, this breaks compatibility on Python3 due to https://docs.python.org/3/whatsnew/3.0.html#ordering-comparisons
+        result = sorted(result, key=lambda obj: obj.name)
```

Comment:
Assuming `result` is an actual list, it's preferred to use `result.sort` rather than creating a new list.